### PR TITLE
Add support for struct.iter_unpack

### LINKED ISSUE #675: Add support for struct.iter_unpack
Should also implement `struct.Struct.iter_unpack` at the same time.

See https://docs.python.org/3/library/struct.html#struct.iter_unpack for details.

Tests in `CPython.test_struct`.

### PR DESCRIPTION
**Pull request by in-code-i-trust**
...and this should resolve #675.

### CODE REVIEW

**Comment by in-code-i-trust** [comment]
The reason of CI failures was the test for `iter_unpack` had a dependency on 'n'/'N' formats.

File: Src/IronPython.Modules/_struct.cs [hunk 1]
```diff
@@ -538,6 +547,72 @@ public class Struct : IWeakReferenceable {
...
+                _owner = owner;
+
+                Reset();
+            }
+
+            public System.Collections.IEnumerator GetEnumerator() {
+                return this;
+            }
+
+            public object Current => _iter_current;
```
Comment (slozier) [inline Src/IronPython.Modules/_struct.cs]: Can you add a `PythonHidden` attribute here so the .NET members don't appear in `dir`? Similaly for `GetEnumerator`, `MoveNext` and `Reset`.
```
>>> dir(struct.iter_unpack("h", b""))
['Current', 'GetEnumerator', 'MoveNext', 'Reset', '__class__', ...
```

```C#
[PythonHidden]
public object Current => _iter_current;
```

File: Src/IronPython.Modules/_struct.cs [hunk 2]
```diff
@@ -538,6 +547,72 @@ public class Struct : IWeakReferenceable {
             #endregion
         }
 
+        [PythonType("unpack_iterator"), Documentation("Represents an iterator returned by _struct.iter_unpack()")]
+        public class PythonUnpackIterator : System.Collections.IEnumerator, System.Collections.IEnumerable {
+            private object _iter_current;
+            private int _next_offset;
+
+            private CodeContext _context;
```
Comment (slozier) [inline Src/IronPython.Modules/_struct.cs]: Maybe mark these as `readonly`?

**Comment by in-code-i-trust** [comment]
> Looks good overall, we should probably check that the buffer is of a correct size like CPython does.
> 
> ```
> >>> import struct
> >>> struct.iter_unpack("h", b"\0")
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
> struct.error: iterative unpacking requires a buffer of a multiple of 2 bytes
> ```

Thank you for reviewing and the comments, fixed in 8830349.

File: Src/IronPython.Modules/_struct.cs (hunk 1 above)
Comment (in-code-i-trust) [reply Src/IronPython.Modules/_struct.cs]: Added the attribute in 8830349.

**Pull request merged**